OpenRA.Mods.Common/UpdateRules/Rules/20210321/AddResourceAccumulator.cs

```diff
@@ -0,0 +1,61 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddResourceAccumulator : UpdateRule
+	{
+		public override string Name => "Refinery had changes to use a ResourceAccumulator trait for altering player resources.";
+
+		public override string Description => "DefaultResourceAccumulator was added and traits UseStorage and DiscardExcessResources were moved from Refinery.";
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			MiniYamlNode resourceAccumulatorNode = null;
+
+			var refineries = actorNode.ChildrenMatching("Refinery");
+			var tiberianSunRefineries = actorNode.ChildrenMatching("TiberianSunRefinery");
```

Comment:
I don't think we support multiple `Refinery` traits on a single actor, so this can be
`var refineryNode = actorNode.LastChildMatching("Refinery") ?? actorNode.LastChildMatching("TiberianSunRefinery");`
and replace the `foreach` below.